### PR TITLE
chore: bump up mathlib to track bitvec-commring

### PR DESCRIPTION
**Pull request by bollu**
This is still WIP, we shall merge this when the PR finished.

### CODE REVIEW

**Comment by bollu** [comment]
It seems our AliveStatements don't compile anymore :( Help from @alexkeizer much appreciated to fix this part.

**Comment by bollu** [comment]
This PR is green now, so review requested @alexkeizer @goens @tobiasgrosser

File: SSA/Projects/InstCombine/AliveStatements.lean [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 open Std
 open Std.BitVec
 open LLVM
 
 @[simp] lemma getLsb_negOne' (w : ℕ) (i : Fin w) :
-    getLsb (-1#w) ↑i :=
-  BitVec.getLsb_negOne ..
+    getLsb (-1#w) ↑i := by 
+  simp only [getLsb_val_eq_getLsb', getLsb'_neg_ofNat_one]
+  -- BitVec.getLsb_negOne ..
```
Comment (goens) [inline SSA/Projects/InstCombine/AliveStatements.lean:16]: huh, what happened here?

File: SSA/Projects/InstCombine/AliveStatements.lean [hunk 2]
```diff
@@ -106,12 +108,12 @@ theorem bitvec_AddSub_1624 :
 theorem bitvec_AndOrXor_135 :
  ∀ (w : Nat) (X C1 C2 : BitVec w), (X ^^^ C1) &&& C2 = X &&& C2 ^^^ C1 &&& C2
 := by alive_auto
-      done --ext
+
```
Comment (goens) [inline SSA/Projects/InstCombine/AliveStatements.lean:111]: some people might complain about the whitespace (I personally don't care :) )

File: SSA/Projects/InstCombine/AliveStatements.lean [hunk 3]
```diff
@@ -500,19 +502,21 @@ theorem bitvec_InstCombineShift__422_1:
 := by alive_auto
       try sorry
 
+-- set_option maxRecDepth 9999 in
 theorem bitvec_InstCombineShift__422_2:
  ∀ (Y X C : BitVec 31), (Y + sshr X (BitVec.toNat C)) <<< C = Y <<< C + X &&& (-1) <<< C
-:= by alive_auto
-      try sorry
+:= by -- alive_auto -- TIMES OUT WITH MAX RECURSION DEPTH
```
Comment (goens) [inline SSA/Projects/InstCombine/AliveStatements.lean:508]: should we open an issue about this? this shouldn't really happen, right?

**Review by goens — APPROVED** [review]
LGTM (left a few minor comments)

**Comment by bollu** [comment]
@goens I fixed the newlines. I'll merge it once CI is green. Speak now or forever hold your peace :)

**Pull request merged**